### PR TITLE
PIL-968 & PIL-984 - BARS integration and error handling

### PR DESCRIPTION
**Pull request by TevynAllen**

### CODE REVIEW

File: test/services/BarsServiceSpec.scala [hunk 1]
```diff
@@ -0,0 +1,557 @@
...
+            request,
+            appConfig(application),
+            messages(application)
+          ).toString
+      }
+    }
+
+  }
+}
+
```
Comment (AKKUM) [inline test/services/BarsServiceSpec.scala:522]: Should we have a test for if any other than NameMatches, SortCodeIsPresentOnEISCD error. If none of the error come , we should check for error page as well.

Comment (TevynAllen) [reply test/services/BarsServiceSpec.scala:522]: I believe all error scenarios are tested based on [the behaviour detailed in the doc](https://confluence.tools.tax.service.gov.uk/pages/viewpage.action?spaceKey=PIL&title=BARS+Integration+and+scenario+handling). For instance some fields that may return an error are not required to be handled in any specific way e.g. `sortCodeSupportsDirectDebit`

File: app/services/BarsService.scala [hunk 2]
```diff
@@ -0,0 +1,259 @@
...
+          accountNumberIsWellFormatted,
+          accountExists,
+          nameMatches,
+          sortCodeIsPresentOnEISCD,
+          userAnswers,
+          mode,
+          form
+        )
+
+      case _ => Future successful Redirect(routes.RepaymentErrorController.onPageLoadError(clientPillar2Id))
```
Comment (AKKUM) [inline app/services/BarsService.scala:150]: Sorry I meant to test this.

**Comment by AKKUM** [comment]
Can we have BankAccountReputation in Service manager config?

File: app/services/BarsService.scala (hunk 2 above)
Comment (TevynAllen) [reply app/services/BarsService.scala:150]: Done

**Review by AKKUM — APPROVED** [review]

**Review by AKKUM — APPROVED** [review]

**Review by AKKUM — APPROVED** [review]

**Review by AKKUM — APPROVED** [review]

**Pull request merged**